zone/bot.cpp

```diff
@@ -9210,3 +9308,2757 @@ void Bot::DoItemClick(const EQ::ItemData *item, uint16 slot_id)
+				return true;
+			}
+
+			return false;
+		case BotSpellTypes::PBAENuke:
+			if (IsPBAENukeSpell(spell_id) && !IsStunSpell(spell_id)) {
+				return true;
+			}
+
+			return false;
+		case BotSpellTypes::AERains:
+			if (IsAERainNukeSpell(spell_id) && !IsStunSpell(spell_id)) {
+				return true;
+			}
+			return false;
+		case BotSpellTypes::AEStun:
+		case BotSpellTypes::Stun:
+			if (IsStunSpell(spell_id)) {
+				return true;
+			}
+
+			return false;
+		case BotSpellTypes::AENukes:
+		case BotSpellTypes::Nuke:
+			if (!IsStunSpell(spell_id)) {
+				return true;
+			}
+
+			return false;
+		default:
+			return true;
+	}
+
+	return true;
+}
+
+void Bot::SetCastedSpellType(uint16 spellType) {
+	_castedSpellType = spellType;
+}
+
+void Bot::DoFaceCheckWithJitter(Mob* tar) {
+	if (!tar) {
+		return;
+	}
+
+	if (IsMoving()) {
+		return;
+	}
+
+	SetCombatJitter();
+	if (!IsFacingMob(tar)) {
+		FaceTarget(tar);
+		return;
+	}
+	return;
+}
+
+void Bot::DoFaceCheckNoJitter(Mob* tar) {
+	if (!tar) {
+		return;
+	}
+
+	if (IsMoving()) {
+		return;
+	}
+
+	if (!IsFacingMob(tar)) {
+		FaceTarget(tar);
+		return;
+	}
+	return;
+}
+
+void Bot::RunToGoalWithJitter(glm::vec3 Goal) {
+	RunTo(Goal.x, Goal.y, Goal.z);
+	SetCombatJitter();
+}
+
+void Bot::SetCombatOutOfRangeJitter() {
+	SetCombatOutOfRangeJitterFlag();
+
+	if (RuleI(Bots, MaxJitterTimer) > 0) {
+		m_combat_jitter_timer.Start(zone->random.Int(RuleI(Bots, MinJitterTimer), RuleI(Bots, MaxJitterTimer)), true);
+	}
+}
+
+void Bot::SetCombatJitter() {
+	SetCombatJitterFlag();
+
+	if (RuleI(Bots, MaxJitterTimer) > 0) {
+		m_combat_jitter_timer.Start(zone->random.Int(RuleI(Bots, MinJitterTimer), RuleI(Bots, MaxJitterTimer)), true);
+	}
+}
+
+void Bot::DoCombatPositioning(
+	Mob* tar,
+	glm::vec3 Goal,
+	bool stopMeleeLevel,
+	float tar_distance,
+	float melee_distance_min,
+	float melee_distance,
+	float melee_distance_max,
+	bool behindMob,
+	bool frontMob
+) {
+	if (HasTargetReflection()) {
+		if (!IsTaunting() && !tar->IsFeared() && !tar->IsStunned()) {
+			if (TryEvade(tar)) {
+				return;
+			}
+		}
+		else if (tar->IsRooted() && !IsTaunting()) { // Move non-taunters out of range - Above already checks if bot is targeted, otherwise they would stay
+			if (tar_distance <= melee_distance_max) {
+				if (PlotBotPositionAroundTarget(tar, Goal.x, Goal.y, Goal.z, (melee_distance_max + 1), (melee_distance_max * 2), false, IsTaunting())) {
+					RunToGoalWithJitter(Goal);
+
+					return;
+				}
+			}
+		}
+		else if (IsTaunting() && ((tar_distance < melee_distance_min) || !frontMob)) { // Back up any bots that are too close
+			if (PlotBotPositionAroundTarget(tar, Goal.x, Goal.y, Goal.z, melee_distance_min, melee_distance, false, IsTaunting())) {
+				RunToGoalWithJitter(Goal);
+
+				return;
+			}
+		}
+	}
+	else {
+		if (!tar->IsFeared()) {
+			if (IsTaunting()) { // Taunting adjustments
+				Mob* mobTar = tar->GetTarget();
+
+				if (!mobTar || mobTar == nullptr) {
```

Comment:
`!mobTar` and `mobTar == nullptr` should be equivalent checks and can be consolidated to just the `!` check.